ledger/src/blockstore_processor.rs

```diff
@@ -853,17 +853,17 @@ pub fn process_blockstore_from_root(
     // We might be promptly restarted after bad capitalization was detected while creating newer snapshot.
     // In that case, we're most likely restored from the last good snapshot and replayed up to this root.
     // So again check here for the bad capitalization to avoid to continue until the next snapshot creation.
-    if !bank_forks
-        .read()
-        .unwrap()
-        .root_bank()
-        .calculate_and_verify_capitalization(debug_verify)
-    {
-        return Err(
-            BlockstoreProcessorError::RootBankWithMismatchedCapitalization(
-                bank_forks.read().unwrap().root(),
-            ),
-        );
+
+    let bank = bank_forks.read().unwrap().root_bank();
```

Comment:
is it ok that we aren't holding the read lock here? Probably.

Reply:
Yes, I believe so: `load_frozen_forks()` guarantees that the bank is frozen and `root_bank()` returns an `Arc<Bank>` so our ref is valid regardless of what `bank_forks` may do with that bank.